### PR TITLE
Static classes

### PR DESCRIPTION
**Pull request by lofcz**
- [x] all members must be static
- [x] can't be instantiated
- [x] can't be used as a base class
- [x] can't contain a constructor

Also adds missing `LoopChk` and `BaseChk` to `InstructionFieldUsage`, tests expected to crash no longer have to provide a blank file with expected stdout.

~~Test 23 is currently failing (point 2 above)~~

Fixes a bug introduced in #52, see test 25 for details.

### CODE REVIEW

File: src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs [hunk 1]
```diff
@@ -477,7 +507,16 @@ public override void Compile(FunctionBuilder bc)
                 bc.Emit_Load(classGlobalRef);
                 bc.Emit_IndexSet(0, 0, fn.Key, isNameIndex: true);
             }
-            
+            //static fields
+            foreach (var field in fields.Where(x => x.Value.Flags.HasFlag(MemberModifierFlags.Static)))
+            {
+                bc.Emit_Literal(DynValue.NewString(field.Key));
```
Comment (CallumDev) [inline src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs:513]: Extraneous instruction, can delete

File: src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs [hunk 2]
```diff
@@ -477,7 +507,16 @@ public override void Compile(FunctionBuilder bc)
...
+                field.Value.Expr.CompilePossibleLiteral(bc);
+                bc.Emit_Load(classGlobalRef);
+                bc.Emit_IndexSet(0, 0, field.Key, isNameIndex: true);
```
Comment (CallumDev) [inline src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs:516]: needs a `bc.Emit_Pop();` as IndexSet only pops 1 value, not 2

File: src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs [hunk 3]
```diff
@@ -1788,8 +1795,12 @@ private int ExecIndex(Instruction i, int instructionPtr)
 
 			// stack: base - index
 			bool isNameIndex = i.OpCode == OpCode.IndexN;
+			bool isMultiIndex = i.OpCode == OpCode.IndexL;
 
-			bool isMultiIndex = (i.OpCode == OpCode.IndexL);
+			if (isNameIndex)
```
Comment (CallumDev) [inline src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs]: Empty if statement?

File: src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs [hunk 4]
```diff
@@ -285,6 +286,10 @@ private DynValue Processing_Loop(int instructionPtr, bool canAwait = false)
 							ref var tab = ref m_ValueStack.Peek();
 							if (top.Type != DataType.Table) throw new InternalErrorException("v-stack top NOT table");
 							if (tab.Type != DataType.Table) throw new InternalErrorException("v-stack tab NOT table");
+							if (top.Table.ModifierFlags.HasFlag(MemberModifierFlags.Static))
```
Comment (CallumDev) [inline src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs:289]: Guard against `Get("Name").String` returning null with a `?? "(null))`. It technically shouldn't be possible, but we don't want to cause a NullReferenceException while having another exception

File: src/WattleScript.Interpreter/Extensions.cs [hunk 5]
```diff
@@ -28,5 +30,12 @@ public static string GetSourceFragment(this SourceRef[] refs, string fullSourceC
 
             return fullSourceCode.Substring(firstNotNull?.FromCharIndex ?? 0, lastNotNull?.ToCharIndex - firstNotNull?.FromCharIndex ?? 0);
         }
+        
+        public static string GetDescription<T>(this T source)
```
Comment (CallumDev) [inline src/WattleScript.Interpreter/Extensions.cs:34]: I get the idea behind this, but this reflection may break a couple of consumers using Unity IL2CPP (unconditional reflection at runtime), and would be best served with a `ToString().ToLowerInvariant()` as the simplest solution for now

**Review by CallumDev — CHANGES_REQUESTED** [review]
Most of the way there, a couple of small changes - and unfortunately I think `GetDescription<T>` breaks a couple of use cases that I am aware wattlescript is currently being used for.

File: src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs (hunk 1 above)
Comment (lofcz) [reply src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs]: fixed in 277b828

File: src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs (hunk 3 above)
Comment (lofcz) [reply src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs]: fixed in 277b828, used as a breakpoint, accidental commit

File: src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs (hunk 4 above)
Comment (lofcz) [reply src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs:289]: fixed in 277b828

File: src/WattleScript.Interpreter/Extensions.cs (hunk 5 above)
Comment (lofcz) [reply src/WattleScript.Interpreter/Extensions.cs]: fixed in 277b828

**Review by CallumDev — APPROVED** [review]

**Pull request merged**